### PR TITLE
store/tikv: define KVFilter for customized mutation initialization

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

Part of #22513

`tablecodec.IsUntouchedIndexKValue` is a TiDB specific optimization. It should not be used in the `store/tikv` package.
For such custom kv initialization requirements, a `KVFilter` is defined for TiKV client users.

### Check List 

Tests 

- Unit test

### Release note 

- `No release note`

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -339,11 +346,15 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 			lockCnt++
 		} else {
 			value = it.Value()
+			isUnnecessaryKV := filter != nil && filter.IsUnnecessaryKeyValue(key, value, flags)
 			if len(value) > 0 {
-				if tablecodec.IsUntouchedIndexKValue(key, value) {
+				if isUnnecessaryKV {
```
Comment (AndreMouche) [inline store/tikv/2pc.go:351]: Only un-empty value need the filter, so could we move the filter logic in L350~L351?

**Review by AndreMouche — COMMENTED** [review]
Rest LGTM

**Comment by sticnarf** [comment]
Seems the test failed, let me see what is wrong...

**Review by AndreMouche — APPROVED** [review]
LGTM

**Comment by disksing** [comment]
@sticnarf CI fail.

**Pull request merged**